DistributionUpdate/PackageUpdate/PackageInfoTools.g

```diff
@@ -336,23 +336,46 @@ UpdatePackageInfoFiles := function(pkgdir)
                 "\n  This is not allowed, so the info file will not be changed\n");
           has_error := true;
         fi;
-        if IsBound(infon.Date) then
-          date := infon.Date;
-          if not IsString(date) and Length(date) = 10 and date{[3,6]} = "//" and
-            ForAll( date{ [1,2,4,5,7,8,9,10] }, IsDigitChar ) then
-            Print("  ERROR (", info.PackageName, "): the date ", date,
-                  "should be a string of the form `dd/mm/yyyy'\n" );
-            has_error := true;
-          else
-            date := List( SplitString( date, "/" ), Int);
-            if not date[1] in [1..31] and
-                   date[2] in [1..12] and
-                   date[3] > 1999 and # GAP 4 appeared in 1999
-                   date[1] > DaysInMonth( date[2], date[3] ) then
-              Print("  ERROR (", info.PackageName, "): the date ", date,
-                    " is invalid\n" );
+        if IsBound( infon.Date ) then
+          if IsString( infon.date ) and Length( infon.Date ) = 10 
+            and infon.Date{ [3,6] } = "//" 
+            and ForAll( infon.Date{ [1,2,4,5,7,8,9,10] }, IsDigitChar ) then
+            date := List( SplitString( infon.Date, "/" ), Int ); # here date=[dd,mm,yyyy]
+            # the format dd/mm/yyyy is ambigous and can be confused with mm/dd/yyyy 
+            # if it is clear from the date that the format is mm/dd/yyyy 
+            # print a message
+            if date[2] in [13..31] and date[1] in [1..12] then
+              Print("  ERROR (", info.PackageName, "): the date ", infon.Date,
+                    Concatenation("should be a string of the form `yyyy-mm-dd`",
+                     "or `dd/mm/yyyy' representing a date since 1999\n" );
               has_error := true;
+              Unbind( date ); 
+              # so that we can check wheter an error was found by checking if
```

Comment:
```suggestion
              # so that we can check whether an error was found by checking if
```
It was not immediately clear for me that this comment refers to the line before. I usually put those comments *before* the code in question, but I know this is not universal. Still, I'd do that, but also rephrase, perhaps like this:
```gap
  # unbind date so that later we can check ...
```